asreview/webapp/authentication/decorators.py

```diff
@@ -81,3 +90,32 @@ def decorated_function(*args, **kwargs):
         return f(projects, *args, **kwargs)
 
     return decorated_function
+
+
+def login_remote_user(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        remote_user_handler = current_app.config.get("REMOTE_USER", False)
+
+        if isinstance(remote_user_handler, RemoteUserHandler):
+            user_info = remote_user_handler.handle_request(request.environ)
+
+            if user_info["identifier"]:
+                user = User.query.filter(
+                    User.identifier == user_info["identifier"]
+                ).one_or_none()
+                if not user:
+                    try:
+                        user = User(
+                            **user_info, origin="remote", public=True, confirmed=True
+                        )
+                        DB.session.add(user)
+                        DB.session.commit()
+                    except (IntegrityError, SQLAlchemyError) as e:
+                        DB.session.rollback()
+                        error_500(e)
```

Comment:
This helper is available in `asreview.webapp.app`, but not in this module or the API. Should `error_500` be moved to a common utility module?

Reply:
That makes sense indeed!

Reply:
You seem to be right. It looks like Flask has simplified the error handling https://flask.palletsprojects.com/en/stable/errorhandling/. Feel free to change it in this PR if that helps.

Reply:
Done.